### PR TITLE
[Fixes] Improved compatibility with Linux & improved robustness

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Added diverse properties that are required when running CARML on Linux (e.g. `-Force`)
- Improved robustness by adding additional checks for e.g. empty files/arrays
- Added support for Test-Local script to support execution without any parameter files

| Pipeline |
| - |
| [AnalysisServices: Servers](https://github.com/Azure/ResourceModules/actions/workflows/ms.analysisservices.servers.yml) |

# Type of Change

Please delete options that are not relevant.

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: utilities/tools/Set-ModuleReadMe.ps1 [hunk 1]
```diff
@@ -845,10 +845,14 @@ function Set-ModuleReadMe {
     $TemplateFilePath = Resolve-Path -Path $TemplateFilePath -ErrorAction Stop
 
     if (-not $TemplateFileContent) {
-        if ((Split-Path -Path $TemplateFilePath -Extension) -eq '.bicep') {
-            $templateFileContent = az bicep build --file $TemplateFilePath --stdout | ConvertFrom-Json -AsHashtable
+        if (-not (Test-Path $TemplateFilePath -PathType 'Leaf')) {
+            throw "[$TemplateFilePath] is no valid file path."
```
Comment (ahmadabdalla) [inline utilities/tools/Set-ModuleReadMe.ps1:849]: is not a valid file path

Comment (MrMCake) [reply utilities/tools/Set-ModuleReadMe.ps1]: Fixed

File: utilities/tools/Test-ModuleLocally.ps1 [hunk 2]
```diff
@@ -279,9 +269,9 @@ function Test-ModuleLocally {
             } catch {
                 Write-Error $_
             } finally {
-                # Restore parameter files
-                # -----------------------
-                if (($ValidationTest -or $DeploymentTest) -and $ValidateOrDeployParameters) {
+                # Restore test files
+                # ------------------
+                if (($ValidationTest -or $DeploymentTest) -and $ValidateOrDeployParameters -and $moduleTestFiles) {
```
Comment (ahmadabdalla) [inline utilities/tools/Test-ModuleLocally.ps1:274]: Isn't $moduleTestFiles implicit? As in it always has a value?

Comment (MrMCake) [reply utilities/tools/Test-ModuleLocally.ps1]: Fair point. Removed

**Review by ahmadabdalla — APPROVED** [review]

**Pull request merged**